src/nvim/window.c

```diff
@@ -4656,7 +4656,11 @@ void win_size_restore(garray_T *gap)
     {
       int i = 0;
       FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
-        frame_setwidth(wp->w_frame, ((int *)gap->ga_data)[i++]);
+        if (wp->w_floating) {
+          win_setwidth_win(((int *)gap->ga_data)[i++], wp);
```

Comment:
This is to save/restore state of split layout when opening temporary split. So we don't need to do anything at all for floats here.